batch/sql/estimated-current.sql

```diff
@@ -278,6 +320,7 @@ CREATE INDEX `jobs_batch_id_state_always_run_cancelled` ON `jobs` (`batch_id`, `
 CREATE INDEX `jobs_batch_id_update_id` ON `jobs` (`batch_id`, `update_id`);
 CREATE INDEX `jobs_batch_id_always_run_n_regions_regions_bits_rep_job_id` ON `jobs` (`batch_id`, `always_run`, `n_regions`, `regions_bits_rep`, `job_id`);
 CREATE INDEX `jobs_batch_id_ic_state_ar_n_regions_bits_rep_job_id` ON `jobs` (`batch_id`, `inst_coll`, `state`, `always_run`, `n_regions`, `regions_bits_rep`, `job_id`);
+CREATE INDEX `jobs_batch_id_job_group_id` ON `jobs` (`batch_id`, `job_group_id`);
 
 CREATE TABLE IF NOT EXISTS `jobs_telemetry` (
   `batch_id` BIGINT NOT NULL,
```

Comment:
Down in the unchanged code, `job_parents` indicates that jobs can have parents in different groups. Verify I understand correctly:

> This is fine because, by design, a job can be cancelled while its children are still Pending. The cancellation loop will eventually notice the cancelled parent and mark the children as cancelled. A cancelled job is only concerned with its member jobs (the jobs with its id as their job_group_id).

Reply:
Not sure who wrote that statement whether it was you or me. That is correct except the last statement should be 

> A cancelled job is only concerned with its child jobs (the jobs with its id as their parent_id). 

Job groups have nothing to do with individual job level cancellation. They are only necessary for a bulk cancellation operation in which case any cancelled jobs will propogate that cancellation event down to their children.

Reply:
er, sorry I meant to say:
> A cancelled job group is only concerned with its member jobs (the jobs with its id as their job_group_id).

And maybe I should have added:
> A cancelled job group need not care about the descendants of its jobs. The job cancelling loop will handle them.